### PR TITLE
common/lockdep: increase MAX_LOCKS to 128k

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/common/lockdep.cc [hunk 1]
```diff
@@ -68,7 +72,7 @@ void lockdep_register_ceph_context(CephContext *cct)
     if (!free_ids_inited) {
       free_ids_inited = true;
       // FIPS zeroization audit 20191115: this memset is not security related.
```
Comment (ronen-fr) [inline src/common/lockdep.cc:74]: probably want to remove the FIPS note

File: src/common/lockdep.cc [hunk 2]
```diff
@@ -94,8 +98,11 @@ void lockdep_unregister_ceph_context(CephContext *cct)
     lock_names.clear();
     lock_ids.clear();
     // FIPS zeroization audit 20191115: these memsets are not security related.
```
Comment (ronen-fr) [inline src/common/lockdep.cc:100]: FIPS note here, too

**Review by ronen-fr — APPROVED** [review]

**Review by lxbsz — APPROVED** [review]

**Comment by lxbsz** [comment]
jenkins test make check

File: src/common/lockdep.cc (hunk 1 above)
Comment (tchaikov) [reply src/common/lockdep.cc:74]: `free_ids.set()` is practically identical to `memset()`. i don't think we should drop this comment.

**Comment by tchaikov** [comment]
changelog:

- remove an empty line.

File: src/common/lockdep.cc (hunk 1 above)
Comment (ronen-fr) [reply src/common/lockdep.cc:74]: I got the impression that the comment was related to an automated search for the text 'memset' that
was intended to identify memsets in specific scenarios. SO probably not relevant to any other clearing
mechanisms.

In any case - a very minor issue. Your call.

Comment (tchaikov) [reply src/common/lockdep.cc:74]: @rzarzynski what do you think? is it safe to drop it and another comment?

**Comment by tchaikov** [comment]
changelog

- do not use `std::for_each_n()` . as libstdc++ shipped along with GCC-8 does not have it.

File: src/common/lockdep.cc (hunk 1 above)
Comment (tchaikov) [reply src/common/lockdep.cc:74]: @ronen-fr removed and repushed.

**Review by ronen-fr — APPROVED** [review]

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
https://pulpito.ceph.com/pdonnell-2021-03-09_03:27:39-fs-wip-pdonnell-testing-20210308.214827-distro-basic-smithi/

- https://tracker.ceph.com/issues/49500
    qa: "Assertion `cb_done' failed."
- https://tracker.ceph.com/issues/48805
    mds: "cluster [WRN] Scrub error on inode 0x1000000039d (/client.0/tmp/blogbench-1.0/src/blogtest_in) see mds.a log and `damage ls` output for details"
- https://tracker.ceph.com/issues/48773
    qa: scrub does not complete
- https://tracker.ceph.com/issues/45434
    qa: test_full_fsync (tasks.cephfs.test_full.TestClusterFull) failed
- https://tracker.ceph.com/issues/49240
    terminate called after throwing an instance of 'std::bad_alloc'
- https://tracker.ceph.com/issues/49466
    qa: "Command failed on gibba030 with status 1: 'set -ex\nsudo dd of=/tmp/tmp.ZEeZBasJer'"
- https://tracker.ceph.com/issues/49684
    qa: fs:cephadm mount does not wait for mds to be created
- https://tracker.ceph.com/issues/48771
    qa: iogen: workload fails to cause balancing

**Comment by batrick** [comment]
CephFS is good. Leave merging to you @tchaikov.

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-03-09_12:22:01-rados-wip-kefu-testing-2021-03-09-1847-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-03-10_06:55:19-rados-wip-kefu-testing-2021-03-09-1847-distro-basic-gibba/

the crash of 

- https://pulpito.ceph.com/kchai-2021-03-10_06:55:19-rados-wip-kefu-testing-2021-03-09-1847-distro-basic-gibba/5952985/
- https://pulpito.ceph.com/kchai-2021-03-10_06:55:19-rados-wip-kefu-testing-2021-03-09-1847-distro-basic-gibba/5952993/

are new ones. but they are not likely to be related to this change.

**Pull request merged**